src/Test.UnitTests.Sarif/UriConverterTests.cs

```diff
@@ -76,5 +83,72 @@ public void WriteJson_ConvertsUriObjectsToStrings()
 
             JsonConvert.SerializeObject(testObject, settings).Should().Be(s_testFileText);
         }
+
+        [Theory]
```

Comment:
>Theory [](start = 9, length = 6)

Rather than `Theory`, which in the past (at least) has been slow, we have an established pattern of looping over test cases and accumulating the failures to report at the end. For an example of the pattern, see src\Test.FunctionalTests.Sarif\SarifValidatorTests.cs.

If you have more recent experience of `Theory`, please let us know.

Reply:
I didn't use Theory initially, but really liked it for telling me which cases were broken (instead of stopping the test on the first failed case).

Runtime is 47ms (Debug) versus 17ms as subsequent calls in the method.

I've changed it to accumulate non-matches. I wish the SDK unit tests didn't take 60 seconds to run, so might as well do my part to help...